client/src/app/core/ui-services/voting.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+
+import { TranslateService } from '@ngx-translate/core';
+
+import { BasePollWithoutNestedModels, PollState, PollType } from 'app/shared/models/poll/base-poll';
+import { ViewAssignmentPoll } from 'app/site/assignments/models/view-assignment-poll';
+import { ViewMotionPoll } from 'app/site/motions/models/view-motion-poll';
+import { AssignmentPollRepositoryService } from '../repositories/assignments/assignment-poll-repository.service';
+import { BannerDefinition, BannerService } from './banner.service';
+import { MotionPollRepositoryService } from '../repositories/motions/motion-poll-repository.service';
+import { OpenSlidesStatusService } from '../core-services/openslides-status.service';
+import { OperatorService } from '../core-services/operator.service';
+
+export enum VotingError {
+    POLL_WRONG_STATE = 1, // 1 so we can check with negation
+    POLL_WRONG_TYPE,
+    USER_HAS_NO_PERMISSION,
+    USER_NOT_PRESENT,
+    USER_HAS_VOTED
+}
+
+interface BannerDefinitionMap {
+    [key: number]: BannerDefinition;
+}
+type ViewAnyPoll = ViewMotionPoll | ViewAssignmentPoll;
+
+@Injectable({
+    providedIn: 'root'
+})
+export class VotingService {
+    // protected so tslint doesn't complain
+    protected motionPollBanners: BannerDefinitionMap = {};
+    protected assignmentPollBanners: BannerDefinitionMap = {};
+
+    public constructor(
+        private operator: OperatorService,
+        motionPollRepo: MotionPollRepositoryService,
+        assignmentPollRepo: AssignmentPollRepositoryService,
+        private banner: BannerService,
+        private translate: TranslateService,
+        private OSStatus: OpenSlidesStatusService
+    ) {
+        motionPollRepo.getViewModelListObservable().subscribe(polls => this.checkForVotablePolls(polls, 'motion'));
+        assignmentPollRepo
+            .getViewModelListObservable()
+            .subscribe(polls => this.checkForVotablePolls(polls, 'assignment'));
+    }
+
+    /**
+     * checks all polls for votable ones and displays banners for them
+     * @param polls the updated poll list
+     * @param mode motion or assignment
+     */
+    private checkForVotablePolls(polls: ViewAnyPoll[], mode: 'motion' | 'assignment'): void {
+        if (this.OSStatus.isInHistoryMode) {
+            polls = []; // display no active polls if in history mode
```

Comment:
please do the cleanup here and finish the block with an return. So it is clear, that the special case was fully handled.